benchmark_stream_test.go

```diff
@@ -0,0 +1,255 @@
+		expectedServerSentStreamMessages     int32
+		expectedServerReceivedStreamMessages int32
+	}{
+		{
+			name:                                 "client stream success",
+			procedure:                            "Bar::ClientStream",
+			method:                               "Bar/ClientStream",
+			numConnections:                       10,
+			requests:                             [][]byte{nil, nil},
+			returnOutput:                         []simple.Foo{{}},
+			expectedInput:                        []simple.Foo{{}, {}},
+			expectedStreamsOpened:                10,
+			expectedServerReceivedStreamMessages: 20,
+			expectedServerSentStreamMessages:     10,
+		},
+		{
+			name:                                 "bidirectional stream success",
+			procedure:                            "Bar::BidiStream",
+			method:                               "Bar/BidiStream",
+			numConnections:                       10,
+			requests:                             [][]byte{nil, nil},
+			returnOutput:                         []simple.Foo{{}, {}},
+			expectedInput:                        []simple.Foo{{}, {}},
+			expectedStreamsOpened:                10,
+			expectedServerSentStreamMessages:     20,
+			expectedServerReceivedStreamMessages: 20,
+		},
+		{
+			name:                                 "server stream success",
+			procedure:                            "Bar::ServerStream",
+			method:                               "Bar/ServerStream",
+			numConnections:                       10,
+			requests:                             [][]byte{nil},
+			returnOutput:                         []simple.Foo{{}},
+			expectedInput:                        []simple.Foo{{}},
+			expectedStreamsOpened:                10,
+			expectedServerSentStreamMessages:     10,
+			expectedServerReceivedStreamMessages: 10,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			svc := &simpleService{
+				expectedInput: tt.expectedInput,
+				returnOutput:  tt.returnOutput,
+			}
+			lis, server := setupGRPCServer(t, svc)
+			defer server.Stop()
+
+			req := &transport.Request{
+				TargetService: "foo",
+				Method:        tt.procedure,
+				Timeout:       time.Second,
+			}
+
+			source, err := protobuf.NewDescriptorProviderFileDescriptorSetBins("./testdata/protobuf/simple/simple.proto.bin")
+			require.NoError(t, err)
+
+			serializer, err := encoding.NewProtobuf(tt.method, source)
+			require.NoError(t, err)
+
+			bench := benchmarkStreamMethod{
+				serializer:            serializer,
+				streamRequest:         &transport.StreamRequest{Request: req},
+				streamRequestMessages: tt.requests,
+			}
+
+			transports, err := warmTransports(bench, tt.numConnections, TransportOptions{
+				ServiceName: "foo",
+				CallerName:  "test",
+				Peers:       []string{"grpc://" + lis.String()},
+			}, _resolvedGrpcProto, 1)
+			require.NoError(t, err)
+
+			for i, transport := range transports {
+				assert.NotNil(t, transport, "transports[%d] must not be nil", i)
```

Comment:
nit: can we use a better indicator than index here to uniquely identify the nil transport? like name or something

Reply:
warmTransports creates the set of transports for the given endpoint, if the transport is nil I don't think we have any identifier except it's index?